pkg/controller/windowsmachineconfig/tracker.go

```diff
@@ -79,14 +75,110 @@ func newNodeRecord(ipAddress string, secretName string) nodeRecord {
 	}
 }
 
+// getWindowsVM returns a windowsVM interface to be used for running commands against
+func getWindowsVM(ipAddress, instanceID string, credentials Credentials) (types.WindowsVM, error) {
+	winVM := &types.Windows{}
+	windowsCredentials := types.NewCredentials(instanceID, ipAddress, credentials.Password, credentials.Username)
+	winVM.Credentials = windowsCredentials
+	// Set up Winrm client
+	if err := winVM.SetupWinRMClient(); err != nil {
+		return nil, errors.Wrap(err, "error instantiating winrm client")
+	}
+	// Set up SSH client
+	if err := winVM.GetSSHClient(); err != nil {
+		return nil, errors.Wrap(err, "error instantiating ssh client")
+	}
+	return winVM, nil
+}
+
+// getInstanceIP gets the instance IP address associated with a node
+func getInstanceIP(nodeList *v1.NodeList, instanceID string) (string, error) {
+	// Ignore the nodes that are not ready.
+	for _, node := range nodeList.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type != v1.NodeReady {
+				continue
+			}
+		}
+		if strings.Contains(node.Spec.ProviderID, instanceID) {
+			for _, address := range node.Status.Addresses {
+				// If external ip exists return it as it is used in AWS, if it doesn't return internal(in case of azure)
+				// TODO: collect all the ips and define a priority order, external then internal etc.
```

Comment:
Why the TODO? Why not do it in this ticket?

If its too much work for this ticket, it needs a link to the ticket this will be handled in.

Reply:
I think whoever is doing the azure work should do it, this is just a hint. Not sure, if we need a ticket for this as this is implementation detail.

Reply:
Please add this as a comment in [WINC-355](https://issues.redhat.com/browse/WINC-355) clearly explaining what needs to be done and why.

Reply:
Please add a link to this discussion on Jira

Reply:
Done, please look at the link below.